Source/BoundaryConditions/PML.cpp

```diff
@@ -491,19 +491,22 @@ PML::PML (const int lev, const BoxArray& grid_ba, const DistributionMapping& /*g
     // Note that this is okay to build pml inside domain for a single patch, or joint patches
     // with same [min,max]. But it does not support multiple disjoint refinement patches.
     Box domain0 = grid_ba.minimalBox();
-    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
-        if (do_pml_Lo[idim]){
-            domain0.growLo(idim, -ncell);
-        }
-        if (do_pml_Hi[idim]){
-            domain0.growHi(idim, -ncell);
+    if (do_pml_in_domain) {
+        for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+            if (do_pml_Lo[idim]){
+                domain0.growLo(idim, -ncell);
+            }
+            if (do_pml_Hi[idim]){
+                domain0.growHi(idim, -ncell);
+            }
```

Comment:
One question to make sure I understand what is being done here (it wasn't added in this PR, but still I'd like to ask). My understanding is that this is the case where we want the PML to overlap with the last `ncell` of the regular domain. Shouldn't we then *add* `ncell` (i.e. grow by `+ncell`, instead of `-ncell`) to the low index and *subtract* `ncell` from the high index (i.e. grow by `-ncell`, as already done here)? If `ncell = 10`, the low index would have to be `0` instead of `-10` (so `-10 + (+ncell) = -10 + (+10) = 0`), such that the PML overlaps with the cells `0:9` of the regular domain rather than spanning the cells `-10:-1` outside of the domain. While when I read `domain0.growLo(idim, -ncell)` it seems like we are lowering the low index even more. I guess I'm missing either the starting point or what `growLo` does precisely. Could you clarify this for me? Thanks!

Reply:
`growLo` means growing in the direction that  points to the lo (i.e., -inf) direction.